Library/Homebrew/cli/parser.rb

```diff
@@ -533,8 +533,8 @@ def formulae(argv)
 
     class OptionConstraintError < UsageError
       def initialize(arg1, arg2, missing: false)
-        arg1 = "--#{arg1.tr("_", "-")}"
-        arg2 = "--#{arg2.tr("_", "-")}"
+        arg1 = dashes(arg1) + arg1.tr("_", "-")
+        arg2 = dashes(arg2) + arg2.tr("_", "-")
```

Comment:
You might be able to use the [`name_to_option`](https://github.com/Homebrew/brew/blob/27afcf577989f9cbbc518bad3ebc15818555b328/Library/Homebrew/cli/parser.rb#L232-L238) method here

Reply:
Hmm, wait. It looks like this is already being called...

https://github.com/Homebrew/brew/blob/27afcf577989f9cbbc518bad3ebc15818555b328/Library/Homebrew/cli/parser.rb#L455

Maybe it doesn't do what I thought...

Reply:
The `tr("_", "-")` part is not correct for e.g. `--screen_saverdir`.

Reply:
Huh. That must have been broken before now. Can you give me a full example command that uses that so I can try to figure it out?

Reply:
@issyl0 I think this will work for the [`check_constraints` method](https://github.com/Homebrew/brew/blob/27afcf577989f9cbbc518bad3ebc15818555b328/Library/Homebrew/cli/parser.rb#L431-L440) which will remove the need for this:

```ruby
      def check_constraints
        @constraints.each do |primary, secondary, constraint_type|
          primary_passed = option_passed?(primary)
          secondary_passed = option_passed?(secondary)
          if :mandatory.equal?(constraint_type) && primary_passed && !secondary_passed
            raise OptionConstraintError.new(name_to_option(primary), name_to_option(secondary))
          end
          raise OptionConstraintError.new(name_to_option(primary), name_to_option(secondary), missing: true) if secondary_passed && !primary_passed
        end
      end
```

Reply:
I think the issue @reitermarkus raised was pre-existing as I didn't change any of the code for the `arg.tr("_", "-")` stuff, so we can tackle that in another PR!

Reply:
It looks like these are the options that will have this issue:

- `--input_methoddir`
- `--internet_plugindir`
- `--audio_unit_plugindir`
- `--vst_plugindir`
- `--vst3_plugindir`
- `--screen_saverdir`

They're all cask options. We may want to check in `name_to_option` to see if the option matched one of those. If so, don't do the `tr`. Separate PR makes sense I think